lib/src/code_field/text_editing_value.dart

```diff
@@ -88,4 +90,91 @@ extension MyTextEditingValue on TextEditingValue {
       selection: TextSelection.collapsed(offset: rangeAfter.start),
     );
   }
+
+  TextEditingValue tabsToSpaces(int spaceCount) {
+    final replacedBeforeSelection = beforeSelection.tabsToSpaces(spaceCount);
+    final replacedInSelection = inSelection.tabsToSpaces(spaceCount);
+    final replacedAfterSelection = afterSelection.tabsToSpaces(spaceCount);
+
+    final finalText =
+        replacedBeforeSelection + replacedInSelection + replacedAfterSelection;
+
+    return TextEditingValue(
+      text: finalText,
+      selection: _getSelectionFromSubstrings(
+        replacedBeforeSelection,
+        replacedInSelection,
+        replacedAfterSelection,
+      ),
+      composing: composing,
+    );
+  }
+
+  TextSelection _getSelectionFromSubstrings(
+    String beforeSelection,
+    String inSelection,
+    String afterSelection,
+  ) {
+    if (selection.baseOffset == -1 || selection.extentOffset == -1) {
+      return const TextSelection.collapsed(offset: -1);
+    }
+
+    final baseOffset = beforeSelection.length;
+    final extentOffset = baseOffset + inSelection.length;
+
+    final result = TextSelection(
```

Comment:
Please use `TextSelection.copyWith` to preserve everything other than the offsets.